src/Storages/MergeTree/MergeTreeDataMergerMutator.h

```diff
@@ -81,7 +81,8 @@ class MergeTreeDataMergerMutator
         bool aggressive,
         size_t max_total_size_to_merge,
         const AllowedMergingPredicate & can_merge,
-        String * out_disable_reason = nullptr);
+        String * out_disable_reason = nullptr,
+        const AllowedSingleMergePredicate & single_merge = [](const MergeTreeData::DataPartPtr &, String *) -> bool { return true; });
```

Comment:
Why we cannot improve our two-parts callback to work with `std::optional<DataPartPtr>` as first argument or something like this?

Reply:
I didn't think of it. Fixed